pkg/common/commands/resetoffset/controller/offsets.go

```diff
@@ -88,38 +81,65 @@ func (r *Reconciler) reconcileOffsets(ctx context.Context, topicName string, gro
 	}
 	logger.Info("Found Topic Partitions", zap.Any("Partitions", partitions))
 
-	// Create An Array Of OffsetMappings To Hold Old/New Offsets
+	// Create An OffsetManager For The Specified ConsumerGroup
+	offsetManager, err := SaramaNewOffsetManagerFromClientFn(groupId, saramaClient)
+	if offsetManager == nil || err != nil {
+		logger.Error("Failed to create OffsetManager for ConsumerGroup", zap.Error(err))
+		return nil, err
+	}
+
+	// Track Partition Specific Resources
+	partitionOffsetManagers := make([]sarama.PartitionOffsetManager, len(partitions))
 	offsetMappings := make([]kafkav1alpha1.OffsetMapping, len(partitions))
 
-	// Loop Over The Partitions - Tracking Status
-	commitOffsets := true
+	// Loop Over The Partitions - Updating Offsets & Tracking Results
+	updatesFailed := false
 	for index, partition := range partitions {
-
-		// Enhance The Logger With Partition
-		partitionLogger := logger.With(zap.Int32("Partition", partition))
-
-		// Update The Partition's Offset (Process all partitions even on error in order to populate OffsetMappings for CRD)
-		offsetMapping, updateErr := updateOffset(partitionLogger, saramaClient, offsetManager, topicName, partition, offsetTime)
+		partitionOffsetManager, offsetMapping, updateErr := updateOffset(logger, saramaClient, offsetManager, topicName, partition, offsetTime)
+		if updateErr != nil {
+			updatesFailed = true
+		}
+		if partitionOffsetManager != nil {
+			partitionOffsetManagers[index] = partitionOffsetManager
+		}
```

Comment:
Since the zero values of the partitionOffsetManagers are nil anyway, this "don't set it if it's nil" logic doesn't really accomplish anything.  Normally I wouldn't mention that here because it doesn't hurt anything, but since you are explicitly setting the value even if there was an error, I might argue that "actually" setting partitionOffsetManagers[index] to nil is the right (philosophically) thing to do here anyway.
```suggestion
		partitionOffsetManagers[index] = partitionOffsetManager
```

Reply:
i can remove the if/nil check - as you noted they are identical.  The code went back and forth a few times between appending to a list vs default sized, etc.. - was just an artifact - good catch ; )